addons/web_editor/static/src/js/backend/html_field.js

```diff
@@ -397,6 +397,10 @@ export class HtmlField extends Component {
                     await this.updateValue();
                 }
                 await saveModifiedImagesPromise;
+                const codeViewEl = this._getCodeViewEl();
+                if (codeViewEl) {
+                    codeViewEl.value = this.wysiwyg.getValue();
+                }
```

Comment:
It's weird to have code regarding the codeview here without testing `this.state.showCodeView`. I suggest to add a check to `this.state.showCodeView`, provided that it still fixes the bug of course.

Reply:
@dmo-odoo `_getCodeViewEl` already checks for `this.state.showCodeView`, whenever we call `_getCodeViewEl` we expect either `false` or the element, do you think we still need to do the check ?

Reply:
Yes, I think so. Even if it is equivalent from a technical, mathematical sense, the two following codes are not equivalent from an idiomatic point of view:
```js
somethingUnrelatedToCodeview();
const codeViewEl = this._getCodeViewEl();
if (codeViewEl) {
    codeViewEl.value = this.wysiwyg.getValue();
}
somethingElseUnrelatedToCodeview();
```
```js
somethingUnrelatedToCodeview();
if (this.state.showCodeView) {
    const codeViewEl = this._getCodeViewEl();
    codeViewEl.value = this.wysiwyg.getValue();
}
somethingElseUnrelatedToCodeview();
```
I personally have a preference for the second one because it looks to make more sense. When you read the first one you think "why do we need to get the codeview here?" and then _sometimes_ `_getCodeViewEl` doesn't get the codeview so... it does nothing. But you have no way to know why sometimes the code does nothing unless you read `_getCodeViewEl` and notice "_oh, ok, it returns nothing when there is no codeview_". This is not obvious from the outside of the function because, for all we know, the codeview element (el) could very much always exist and just be hidden when the code view is not shown. You can't know that without reading `_getCodeViewEl`, something you don't always want to do, and especially not in the middle of a function that does not seem to be related to the codeview.

Reply:
makes sense, thanks for explaining, i agree that sometimes we should go a bit beyond "it works fine"